src/transformers/generation/utils.py

```diff
@@ -1147,6 +1147,21 @@ def _validate_model_kwargs(self, model_kwargs: Dict[str, Any]):
         # `prepare_inputs_for_generation` doesn't accept them, then a stricter check can be made ;)
         if "kwargs" in model_args or "model_kwargs" in model_args:
             model_args |= set(inspect.signature(self.forward).parameters)
+
+        # Generic composite models need information from its 2 components
+        if self.__class__.__name__ in {
+            "EncoderDecoderModel",
+            "VisionEncoderDecoderModel",
+            "SpeechEncoderDecoderModel",
+        }:
+            # allow encoder kwargs
+            encoder_model_args = set(inspect.signature(self.encoder.forward).parameters)
+            model_args |= encoder_model_args
+
+            # allow decoder kwargs
+            decoder_model_args = set(inspect.signature(self.decoder.forward).parameters)
+            model_args |= {f"decoder_{x}" for x in decoder_model_args}
```

Comment:
This `decoder_` prefix here is to correspond to the logic in 

https://github.com/huggingface/transformers/blob/ee4250a35f3bd5e9a4379b4907b3d8f9d5d9523f/src/transformers/models/encoder_decoder/modeling_encoder_decoder.py#L462-L464

Reply:
@gante thing get complicated with Bart family models.

It has `self.model = BartModel(config)`, and the encoder/decoder is in its base model `self.model`.
Let me know if you want me to address this in this PR or we can keep the previous (simple but working) version, and see what we want to do later?

Reply:
It should be fine like this :)